utils/src/types/hint.rs

```diff
@@ -0,0 +1,49 @@
+use crate::buffer::Buffer;
+use crate::decode::*;
+use crate::fixed_size_bytes;
+
+fixed_size_bytes!(Hint, 4);
+
+#[cfg(test)]
+mod tests {
+    extern crate std;
+
+    use super::{new_decoder, Hint};
+    use crate::buffer::Buffer;
+    use crate::decode::Decoder;
+    use crate::TempData;
+    use rand::Rng;
+    use std::vec;
+    use std::vec::Vec;
+
+    fn gen_bytes(min_length: usize, max_length: usize) -> Vec<u8> {
```

Comment:
Seems that we can import this function from `test` module, same with `public_key.rs`